hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/BasicColStat.java

```diff
@@ -0,0 +1,121 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.hiero.sketch.table.api.IColumn;
+import org.hiero.sketch.table.api.IMembershipSet;
+import org.hiero.sketch.table.api.IRowIterator;
+import org.hiero.sketch.table.api.IStringConverter;
+import javax.annotation.Nullable;
+
+public class BasicColStat {
+    private final int MOMENT_NUM;
+    private double min;
+    @Nullable private Object minObject;
+    private double max;
+    @Nullable private Object maxObject;
+    private final double moments[];
+    private long size;
+
+    public BasicColStat(int momentNum) {
+        if (momentNum < 1)
+            throw new IllegalArgumentException("number of moments cannot be positive");
+        this.MOMENT_NUM = momentNum;
+        this.moments = new double[this.MOMENT_NUM];
+        this.min = Double.POSITIVE_INFINITY;
+        this.max = Double.NEGATIVE_INFINITY;
+    }
+
+    public BasicColStat() {
+        this.MOMENT_NUM = 2;
+        this.moments = new double[this.MOMENT_NUM];
+        this.min = Double.POSITIVE_INFINITY;
+        this.max = Double.NEGATIVE_INFINITY;
+    }
+
+    public double getMin() { return this.min; }
+
+    public Object getMinObject() {
+        assert this.minObject != null;
```

Comment:
This assert should not be here.
Instead the function should be tagged as @Nullable.